### PR TITLE
Removed support for Custom Keystore provider

### PR DESCRIPTION
**Pull request by yitam**
This change is [Reviewable](https://reviewable.io/reviews/microsoft/msphpsql/682)

### CODE REVIEW

File: test/functional/pdo_sqlsrv/skipif_mid-refactor.inc [hunk 1]
```diff
@@ -1,16 +1,21 @@
...
+    echo("Error: could not connect during SKIPIF!");
+} elseif (isColEncrypted()) {
+    if (!(strtoupper(substr(PHP_OS, 0, 3)) === 'WIN')) {
+        die( "Skip, AE test on windows only." );
+    }
+
+    if (!isAEQualified($conn)) {
+        die("skip - AE feature not supported in the current environment.");
+    }
+}
```
Comment (lilgreenbird) [inline test/functional/pdo_sqlsrv/skipif_mid-refactor.inc]: this is not used elsewhere?

File: test/functional/sqlsrv/skipif_versions_old.inc [hunk 2]
```diff
@@ -1,15 +1,21 @@
...
+} elseif (AE\isColEncrypted()) {
+    if (!isWindows()) {
+        die( "Skip, AE test on windows only." );
+    }
+
+    if (!AE\isQualified($conn)) {
+        die("skip - AE feature not supported in the current environment.");
+    }
+}
 ?>
```
Comment (lilgreenbird) [inline test/functional/sqlsrv/skipif_versions_old.inc]: what is this weird character?!?!

File: test/functional/sqlsrv/skipif_versions_old.inc [hunk 3]
```diff
@@ -1,15 +1,21 @@
...
+    echo("Error: could not connect during SKIPIF!");
+} elseif (AE\isColEncrypted()) {
+    if (!isWindows()) {
+        die( "Skip, AE test on windows only." );
+    }
+
+    if (!AE\isQualified($conn)) {
+        die("skip - AE feature not supported in the current environment.");
+    }
+}
```
Comment (lilgreenbird) [inline test/functional/sqlsrv/skipif_versions_old.inc]: this file seems to be used to skipif_ae_supported

File: test/functional/sqlsrv/sqlsrv_connect_driver.phpt [hunk 4]
```diff
@@ -1,158 +1,163 @@
...
+
+    $connectionOptions['Driver']=$value;
+    $expected = "The specified ODBC Driver is not found.";
+
+    connectVerifyOutput($server, $connectionOptions, $expected);
+}
+
+?>
+--EXPECT--
+Done
```
Comment (lilgreenbird) [inline test/functional/sqlsrv/sqlsrv_connect_driver.phpt]: why were there soo many lines changed when I don't really see any change in most of the lines?!?!

File: test/functional/sqlsrv/sqlsrv_connect_encrypted.phpt [hunk 5]
```diff
@@ -1,102 +1,102 @@
...
+       $expected_substr = "Invalid value type for option ColumnEncryption was specified.  String type was expected.";
+        if( strpos(sqlsrv_errors($conn)[0]['message'], $expected_substr ) === false )
+        {
+            print_r(sqlsrv_errors());
+        }
+    }
+}
+?>
+--EXPECT--
 Done
```
Comment (lilgreenbird) [inline test/functional/sqlsrv/sqlsrv_connect_encrypted.phpt]: again this funny character

**Comment by lilgreenbird** [comment]
We currently always skip AE if not windows.  We need a test that doesn't skip and make sure it handles that nicely.

File: test/functional/pdo_sqlsrv/skipif_mid-refactor.inc (hunk 1 above)
Comment (yitam) [reply test/functional/pdo_sqlsrv/skipif_mid-refactor.inc]: what do you mean?

File: test/functional/sqlsrv/skipif_versions_old.inc (hunk 3 above)
Comment (yitam) [reply test/functional/sqlsrv/skipif_versions_old.inc]: I don't think there's skipif_ae_supported?

File: test/functional/sqlsrv/sqlsrv_connect_driver.phpt (hunk 4 above)
Comment (yitam) [reply test/functional/sqlsrv/sqlsrv_connect_driver.phpt]: sorry I think it's the line ending problem... anyway, the changes are mainly in testEncryptedWithODBC() 
the same applies to the other equivalent test in pdo

File: test/functional/sqlsrv/skipif_versions_old.inc (hunk 3 above)
Comment (lilgreenbird) [reply test/functional/sqlsrv/skipif_versions_old.inc]: I mean, that's what it's really doing

**Pull request merged**